rfc/graphql-schema/additionalTypeDefs/byPage/invoices.ts

```diff
@@ -0,0 +1,143 @@
+export default /* GraphQL */ `
+  enum InvoiceStatus {
+    OPEN
+    PAID
+    OVERDUE
+    CANCELLED
+  }
+
+  type InvoiceStats {
+    openBalance: Float!
+    overdueBalance: Float!
+  }
+
+  input InvoiceDateRangeFilterInput {
+    from: DateTime!
+    to: DateTime
+  }
+
+  input InvoiceFiltersInput {
+    search: String
+    dateRange: InvoiceDateRangeFilterInput
+    status: InvoiceStatus
+  }
+
+  enum InvoicesSortInput {
+    INVOICE_NUMBER_A_TO_Z
+    INVOICE_NUMBER_Z_TO_A
+    ORDER_ID_A_TO_Z
+    ORDER_ID_Z_TO_A
+    CREATED_AT_NEWEST
+    CREATED_AT_OLDEST
+    DUE_DATE_NEWEST
+    DUE_DATE_OLDEST
+    ORIGINAL_BALANCE_HIGHEST
+    ORIGINAL_BALANCE_LOWEST
+    OPEN_BALANCE_HIGHEST
+    OPEN_BALANCE_LOWEST
+  }
+
+  type CreateCartFromInvoiceError implements Error {
+    message: String!
+  }
+
+  type CartCreateResult {
+    cart: Cart
+    errors: [CreateCartFromInvoiceError!]!
+  }
+
+  type GenerateInvoicePdfError implements Error {
+    message: String!
+  }
+
+  type GenerateInvoicePdfResult {
+    url: String
+    errors: [GenerateInvoicePdfError!]!
+  }
+
+  type Invoice {
+    id: ID!
+    invoiceNumber: String!
+    createdAt: DateTime!
+    dueDate: DateTime!
+    openBalance: Money!
+    originalBalance: Money!
+    status: InvoiceStatus!
+    order: Order!
+    company: Company!
+  }
+
+  type InvoiceEdge {
+    node: Invoice!
+    cursor: String!
+  }
+
+  type InvoiceConnection {
+    edges: [InvoiceEdge!]!
+    pageInfo: PageInfo!
+    collectionInfo: CollectionInfo!
+    stats: InvoiceStats!
+  }
+
+  type ExportInvoicesAsCSVResult {
+    url: String
+    errors: [Error!]!
+  }
+
+  extend type Company {
+    invoicesByIds(invoiceIds: [ID!]!): [Invoice!]!
+    invoices(
+      filters: InvoiceFiltersInput
+      sortBy: InvoicesSortInput
+      before: String
+      after: String
+      first: Int
+      last: Int
+    ): InvoiceConnection!
+  }
+
+  input InvoicePaymentInput {
+    id: ID!
+    # If no amount is specified, the whole invoice amount will be used
+    amount: Float
```

Comment:
shame we can't have unions for inputs

Reply:
I know 😔 

I thought of having two different mutations, one to pay in full, one to specify amounts, but it feels a bit extreme. What do you think?

Reply:
yeah, I'm not sure which is best. 

I guess splitting means there's fewer possible error types for the pay in full (as it doesn't need to validate the user's amount input) so it's a simpler signature for a reasonable proportion of mutations.

Would there be any BE benefit to splitting?

---

otherwise

maybe we can name it something more indicative of being less than the full amount?
`shortPayAmount`, `partialAmount`, 🤔 

---

also, should it be our `Money` type?